### ISSUE TITLE
Make test output more readable

### ISSUE THREAD

**Issue by shubheksha** [kubernetes-sigs/service-catalog#2354]
# Bug Report 

**What happened**:
Currently, the text output is really hard to parse. It's a wall of text and isn't colourized.

**What you expected to happen**:
- Highlighting failures would be of immense help in figuring out what's wrong.
- For issues with, for example, a mismatch in golden files, we can link to the appropriate section in the devguide.

**How to reproduce it (as minimally and precisely as possible)**:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/kubernetes-incubator_service-catalog/2353/pull-service-catalog-unit/410

**Anything else we need to know?**:

**Environment**: N/A
- Kubernetes version (use `kubectl version`):
- service-catalog version:
- Cloud provider or hardware configuration:
- Do you have api aggregation enabled? 
  - Do you see the configmap in kube-system? 
  - Does it have all the necessary fields?
    - `kubectl get cm -n kube-system extension-apiserver-authentication -o yaml` and look for `requestheader-XXX` fields
- Install tools:
  - Did you use helm? What were the helm arguments? Did you `--set` any extra values?
- Are you trying to use ALPHA features? Did you enable them?

/help

**Comment by shubheksha** [comment]
@carolynvs mentioned a few ideas for fixing this here: https://github.com/kubernetes-incubator/service-catalog/pull/2353#issuecomment-423205176

**Comment by carolynvs** [comment]
I'm removing the `help wanted` label because we use that label a bit differently on this repo. 

`help wanted` is what new contributors use to find issues that have extra information to help them figure out how to tackle the issue (for example where to find the right code to change, exactly what changes to make and tests to update). At the moment this issue doesn't have that information, but as we figure out what we'd like, we can add that info and then put the label on. 😀 

Here's more doc on how we use the `good first issue` and `help wanted` labels. https://github.com/kubernetes/community/blob/master/contributors/devel/help-wanted.md
